libs/eslint-plugin/docs/rules/no-zone-critical-rxjs-creation-apis.md

```diff
@@ -0,0 +1,51 @@
+# no-zone-critical-rxjs-creation-apis
+
+> Detects Zone critical RxJS creation APIs.
+
+## Rationale
+
+RxJS creation APIs that rely on Zone patched scheduling APIs might cause unnecessary change detection runs.
+
+## Examples
+
+<details>
+<summary>❌ Examples of incorrect code for this rule</summary>
+
+```ts
+import { fromEvent } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+const source = fromEvent(document, 'click');
+const example = source.pipe(map((event) => 'Event time: ' + event.timeStamp));
+const subscribe = example.subscribe((val) => console.log(val));
+```
+
+```ts
+import { interval } from 'rxjs';
+
+const source = interval(1000);
+const subscribe = source.subscribe((val) => console.log(val));
+```
+
+```ts
+import { timer } from 'rxjs';
+
+const source = timer(1000, 2000);
+const subscribe = source.subscribe((val) => console.log(val));
+```
+
+</details>
+
+<br />
+
+<details>
+<summary>✅ Examples of correct code for this rule</summary>
+
+```ts
+import { interval } from '@rx-angular/cdk/zone-less';
+
+const source = interval(1000);
+const subscribe = source.subscribe((val) => console.log(val));
```

Comment:
This one I guess not relevant anymore since we removed zone-less/rxjs and having it as separate deprecated package. Maybe something with getZoneUnpatchedApi will be a better fit

Reply:
Oh right, so the import would have to be from `@rx-angular/cdk/zone-less/rxjs` now? But if that package is deprecated anyway (\*), then what is now the zone-less way to use these RxJS creators/operators/schedulers? I'm struggling to think of a "correct" example otherwise. Does it still make sense to have these zone-less RxJS rules?

---

(\*) BTW if zone-less RxJS is deprecated, then the docs are very misleading. E.g. [zone-less docs still have an RxJS section](https://github.com/rx-angular/rx-angular/tree/main/libs/cdk/zone-less/docs#usage) and [the standalone package docs mention nothing about deprecation either](https://github.com/rx-angular/rx-angular/tree/main/libs/cdk/zone-less/rxjs). I could only find JSDoc `@deprecated will be removed with 1.0.0` comments in the code itself, but that's only because I went looking for it after your comment :sweat_smile:

Reply:
We moved package [here](https://github.com/rx-angular/rxjs-zone-less). Guess we missed docs part. Yeah, as for example no clue honestly. Maybe we don't need this rules anymore. @hoebbelsB @BioPhoton @edbzn what do you think? Should we keep this rules just to "frighten" :D people? Or maybe we can come up with some example?

Reply:
I think the rule is still valid. We need to update the docs and the rule can just point to the other package?

Reply:
Ok, I updated the examples.